src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,149 @@
+import { FC, useEffect, useState } from 'react';
+
+import { useParams, useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../PeopleFilters/PeopleFilters';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+import { Person } from '../../types';
+import { getFullDetailsOfPerson } from '../../utils/helpers';
+import { getPeople } from '../../api';
+import { OrderByType, SortBy } from '../../types/OrderAndSortTypes';
+
+export const PeoplePage: FC = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const { slug = '' } = useParams();
+
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const visiblePeoples = peoples.filter(person => {
+    const {
+      name,
+      sex: personSex,
+      born,
+      father,
+      mother,
+    } = person;
+
+    const visibleBySearch = [name, father?.name, mother?.name]
+      .some(n => n?.toLowerCase().trim().includes(query.toLowerCase()));
+
+    const visibleBySex = sex === '' || personSex === sex;
+
+    const century = Math.ceil(born / 100);
+    const visibleByCenturies = centuries.length === 0
+      || centuries.includes(century.toString());
+
+    return visibleBySearch && visibleBySex && visibleByCenturies;
+  });
+
+  const sortedPeoples = (
```

Comment:
Consider moving the sorting function outside of the component to avoid redeclaring it during every re-render.